templates/IAM/iam-role.yaml

```diff
@@ -0,0 +1,41 @@
+# Define a role
+#
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Creates a role'
+Parameters:
+  AssumeRolePolicyDocument:
+    Type: String
+    Description: Policy describing trust relationships
+    Default: ''
+  ManagedPolicyArns:
+    Type: CommaDelimitedList
+    Default: ''
+    Description: A list of managed policies for the role.
+  MaxSessionDuration:
+    Type: Number
+    Description: Maximum session duration in seconds
+    Default: 43200
+  RoleName:
```

Comment:
in general, it's better to let AWS generate role names otherwise users might create conflicting names without knowing.  At the very least you should make this an optional parameter.